client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -117,7 +117,7 @@
                     handleMissingLog();
                     segmentWriter = selector.getSegmentOutputStreamForKey(routingKey);
                 }
-                segmentWriter.write(new PendingEvent(routingKey, data, ackFuture));
+                segmentWriter.write(PendingEvent.withHeader(routingKey, data, ackFuture));
```

Comment:
For symmetry, should we have an output class like `EventSegmentInputStream` even if short?

Reply:
I suppose we could. It be the same as the existing SegmentOutputStream class, and all methods would delegate to a wrapped instance. Except one that took the three arguments passed here and invoked `PendingEvent.withHeader` before passing the result to the wrapped writer.
It'd add like 300 lines to our code base, for the sake of abstracting out just part of this one line. 
If you really want that, I can add it.

Reply:
It is not a must, but symmetry typically makes the code easier to read. You seem to be on the fence slightly inclined towards not doing it. I'm fine with that assessment.